mypy/test/testconfigparser.py

```diff
@@ -1,168 +1,50 @@
-
-    def test_parse_config_file_with_mixed_valid_and_invalid_entries(self) -> None:
-        """Mix of valid and invalid filenames should raise an error."""
-        with tempfile.TemporaryDirectory() as tmpdirname:
-            config_path = os.path.join(tmpdirname, "test_config.ini")
-
-            with open(config_path, "w") as f:
-                f.write(
-                    """
-                    [mypy]
-                    files = file1.py, , , file2.py
-                    """
-                )
+@pytest.mark.parametrize(
+    "config_content, expected_files, expected_exception",
+    [
+        # Files listed without spaces
+        ("[mypy]\nfiles =file1.py,file2.py,file3.py", ["file1.py", "file2.py", "file3.py"], None),
 
-            options = Options()
+        # Files listed with adequate space
+        ("[mypy]\nfiles =file1.py, file2.py, file3.py", ["file1.py", "file2.py", "file3.py"], None),
 
-            with self.assertRaises(ValueError) as cm:
-                parse_config_file(options, lambda: None, config_path, stdout=None, stderr=None)
+        # Files with extra spaces
+        ("[mypy]\nfiles =  file1.py ,   file2.py  ,   file3.py", ["file1.py", "file2.py", "file3.py"], None),
 
-            self.assertIn("Invalid config", str(cm.exception))
+        # Files listed with a trailing comma
+        ("[mypy]\nfiles =  file1.py, file2.py, file3.py,", ["file1.py", "file2.py", "file3.py"], None),
 
-    def test_parse_config_file_with_newlines_between_files(self) -> None:
-        """Newlines between file entries should be correctly handled."""
-        with tempfile.TemporaryDirectory() as tmpdirname:
-            config_path = os.path.join(tmpdirname, "test_config.ini")
+        # Empty files key
+        ("[mypy]\nfiles =", [], None),
 
-            with open(config_path, "w") as f:
-                f.write(
-                    """
-                    [mypy]
-                    files = file1.py,
-                            file2.py,
-                            file3.py
-                    """
-                )
+        # Files key with only a comma
+        ("[mypy]\nfiles = ,", None, ValueError),
 
-            options = Options()
+        # Mixed valid and invalid filenames
+        ("[mypy]\nfiles = file1.py, , , file2.py", None, ValueError),
 
-            parse_config_file(options, lambda: None, config_path, stdout=None, stderr=None)
+        # Files listed with multiple trailing comma
+        ("[mypy]\nfiles =  file1.py, file2.py, file3.py,", None, ValueError),
 
-            self.assertEqual(options.files, ["file1.py", "file2.py", "file3.py"])
+        # Newlines between file entries
+        ("[mypy]\nfiles = file1.py,\nfile2.py,\nfile3.py", ["file1.py", "file2.py", "file3.py"], None),
+    ]
+)
+def test_parse_config_file(tmp_path, config_content, expected_files, expected_exception):
+    """Parameterized test for parse_config_file handling various configurations."""
```

Comment:
Function docstrings should start with a verb so that they reference the action being performed. Only constants / objects need to be described as terms because they don't do anything.

Additionally, it should probably be more accurate. Generic statements aren't very useful because they don't include any specifics and so it's normally pointless to have them like that.